src/Sentry/SentryStructuredLogger.cs

```diff
@@ -0,0 +1,95 @@
+using Sentry.Infrastructure;
+
+namespace Sentry;
+
+/// <summary>
+/// Creates and sends logs to Sentry.
+/// <para>This API is experimental and it may change in the future.</para>
+/// </summary>
+[Experimental(DiagnosticId.ExperimentalFeature)]
+public abstract class SentryStructuredLogger
+{
+    private protected SentryStructuredLogger()
+    {
+    }
+
+    private protected abstract void CaptureLog(SentryLogLevel level, string template, object[]? parameters, Action<SentryLog>? configureLog);
```

Comment:
Eventually, if the Sentry SDK itself will also create logs, we probably want to make this internal rather than private and add an optional [origin](https://develop.sentry.dev/sdk/telemetry/logs/#sdk-integration-attributes) argument (that defaults to null)... or alternatively we add a bunch of internal overloads to the `LogTrace`, `LogDebug` etc. methods, accepting such a parameter. 

I'm not sure whether we plan to set that origin attribute for integrations like our NLog or Serilog integrations. Those logs do technically come from an integration but ultimately I think they're logs that come from the user.